### PR TITLE
[RHELC-1282] Override yum config exclude list

### PR DESCRIPTION
**Pull request by r0x0d**
Override the exclude list from the yum config file to avoid dependency problems during the transaction process.

This method is temporary as it will only override the config file while running the transaction, no modifications to the config file physically will be done.

Jira Issues: [RHELC-1282](https://issues.redhat.com/browse/RHELC-1282)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [ ] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

**Comment by r0x0d** [comment]
Suggestion from @kokesak in slack: 

>We could simply add the redhat-release-server package to the yum conf exclude list, boot into an older kernel (for example, 3.10.0-1160.95.1.el7.x86_64) and run the transaction in analysis mode. We can check explicitly for not having an occurrence of the following packages in the transaction logs: 
>
>- grubby-8.28-26.el7.x86_64 requires system-release
>- kernel-3.10.0-1160.105.1.el7.x86_64 requires system-release
>- setup-2.8.71-11.el7.noarch requires system-release
>- initscripts-9.49.53-1.el7_9.1.x86_64 requires redhat-release

**Comment by r0x0d** [comment]
This is supposed to land on 1.8.

**Review by pr-watson — APPROVED** [review]
Looks good to me

**Review by SpyTec — APPROVED** [review]

**Comment by r0x0d** [comment]
@danmyway, for some reason, the newly added tests are picking an "older" version instead of what is in the latest commit...

**Comment by danmyway** [comment]
@r0x0d I don't think I follow, do you have an example somewhere?

**Comment by r0x0d** [comment]
> @r0x0d I don't think I follow, do you have an example somewhere?

Nevermind... I think I was just seeing the wrong test.

**Comment by SpyTec** [comment]
@r0x0d can you rebase and fix conflicts?

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -623,3 +623,87 @@ def remediation_out_of_date_packages(shell):
...
     for package in problematic_packages:
         shell(f"yum update -y {package}")
+
+
+@pytest.fixture(scope="function")
+def kernel_check_envar():
+    """
+    Fixture.
+    Set CONVERT2RHEL_UNSUPPORTED_SKIP_KERNEL_CURRENCY_CHECK environment variable
+    to skip the kernel currency check.
```
Comment (SpyTec) [inline tests/integration/conftest.py:633]:
```suggestion
    Fixture.
    Set CONVERT2RHEL_SKIP_KERNEL_CURRENCY_CHECK environment variable
    to skip the kernel currency check.
```

**Pull request merged**